packages/@aws-cdk/aws-codebuild/test/test.project.ts

```diff
@@ -797,6 +798,52 @@ export = {
 
       test.done();
     },
+
+    'certificate arn'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack();
+      const bucket = new s3.Bucket(stack, 'Bucket');
+
+      // WHEN
+      new codebuild.Project(stack, 'Project', {
+        source: codebuild.Source.s3({
+          bucket,
+          path: 'path',
+        }),
+        environment: {
+          certificate: {
+            bucket,
+            objectKey: 'path',
+          },
+        },
+      });
+
+      // THEN
+      expect(stack).to(haveResourceLike('AWS::CodeBuild::Project', {
+        Environment: objectLike({
+          Certificate: {
+            'Fn::Join': [
+              '',
+              [
+                'arn:',
+                {
+                  'Ref': 'AWS::Partition',
+                },
+                ':s3:::',
+                {
+                  'Ref': stack.getLogicalId(bucket.node.defaultChild as CfnBucket),
```

Comment:
No need for these tricks - just use an imported Bucket (`Bucket.fromBucketArn/Name()`) instead of a new one.

Reply:
There is already a bucket used for the project source which I re-used for the IBucket on line 814 as the source bucket for where the certificate is. I can import it vs. creating it if that's what you mean.

The test then asserts that the cloudformation has a project, with a certificate, that uses the ref intrinsic function to get the bucket name using the bucket's logical id which is generated by CDK and I don't know ahead of time.

I have to know the logical ID of the bucket that was used as IBucket interface on line 814. Is there a better way to get the generated Logical ID of the construct? `Bucket.fromBucketArn/Name` is only going to import a new bucket yeah?

Reply:
NM, I see what you mean. If I import the bucket the Logical ID there is no ref and so no need to check the logical ID :). Got it.